grafanalib/core.py

```diff
@@ -551,6 +764,27 @@ def to_json_data(self):
             'type': 'query',
         }
 
+    @staticmethod
+    def parse_json_data(data):
+        if 'current' in data:
+            current = data.pop('current')
+            if 'text' in current:
+                data['default'] = current['text']
+            elif 'value' in current:
+                data['default'] = current['value']
+        if 'datasource' in data:
+            data['dataSource'] = data.pop('datasource')
+
+        # Deleted values
```

Comment:
Why are they deleted?

Reply:
If you take a look at the code above the values were put in the JSON but no attributes were created (used). 
eg. `sort` is set to `1`.
I fixed that.